### PR TITLE
obs-text: Add text transform property

### PR DESCRIPTION
**Pull request by exeldro**
Allows text to be converted to all capital letters
image

https://ideas.obsproject.com/posts/133/option-to-force-all-caps-on-text-sources

### CODE REVIEW

**Comment by WizardCM** [comment]
I'd personally name it "Transform text to uppercase" or something. "All caps" is too informal.

**Comment by exeldro** [comment]
@WizardCM that would give a very long variable name. Or only change it in the locale file?

File: plugins/obs-text/gdiplus/obs-text.cpp [hunk 1]
```diff
@@ -737,6 +743,9 @@ inline void TextSource::Update(obs_data_t *s)
 		if (!text.empty())
 			text.push_back('\n');
 	}
+	if(all_caps)
+		transform(text.begin(), text.end(),
```
Comment (RytoEX) [inline plugins/obs-text/gdiplus/obs-text.cpp]: Style:  This line does not _need_ to have a line-break.
```suggestion
		transform(text.begin(), text.end(), text.begin(), towupper);
```

File: plugins/obs-text/gdiplus/obs-text.cpp [hunk 2]
```diff
@@ -737,6 +743,9 @@ inline void TextSource::Update(obs_data_t *s)
...
+				text.begin(), towupper);
```
Comment (RytoEX) [inline plugins/obs-text/gdiplus/obs-text.cpp]: This line can be unwrapped/removed.
```suggestion
```

**Comment by WizardCM** [comment]
> 
> 
> @WizardCM that would give a very long variable name. Or only change it in the locale file?

You could name the variable `uppercase`, then update the locale.

**Comment by exeldro** [comment]
Done suggested changes by @WizardCM and @RytoEX

**Comment by Andersama** [comment]
@exeldro you might want to change the property to an int and add lowercase as an option.

**Comment by exeldro** [comment]
@Andersama Are there use cases for all lowercase text?

**Comment by WizardCM** [comment]
Mostly comes down to a streamer's style.

**Comment by exeldro** [comment]
Added lowercase option

File: plugins/obs-text/gdiplus/obs-text.cpp [hunk 3]
```diff
@@ -737,6 +751,10 @@ inline void TextSource::Update(obs_data_t *s)
 		if (!text.empty())
 			text.push_back('\n');
 	}
+	if(text_transform == S_TRANSFORM_UPPERCASE)
```
Comment (Andersama) [inline plugins/obs-text/gdiplus/obs-text.cpp]: this can be made into an if else if block, saves on a conditional check

**Comment by SuslikV** [comment]
I think, that "Convert to" is better wording here. Because "Transform" in obs is already used for geometry.

**Pull request merged**